### PR TITLE
[Issue #8969] Ensure save opportunity icon works as expected

### PR DESCRIPTION
**Pull request by freddieyebra**
## Summary

Fixes / Work for #8969 

## Changes proposed

This pr adds logic for ensuring that the save opportunity icon on the search results and opportunity detail page correctly reflects the saved status against the users individually saved opportunities.

This pr also includes methods for determining the scope of the request based on organization_id and scope params. It will only return the default until filters are supported.

## Context for reviewers

Prior to this change, the saved opportunity icon was checking if the opportunity was saved against the "all" scope of saved opportunity instead of the "individual" scope saved opportunities response. This updates those checks to check against the individual saved opportunities.

## Validation steps

1. Ensure db is seeded by running `make db-seed-local`
2. Find the `isShareWithOrganizationEnabled` and set it to true and run the api and frontend
3. Login as two_org_user
4. Navigate to workspaces > organizations and ensure you have 2 organizations listed
5. Navigate to the saved opportunities page and assert there are not saved opportunities
6. Navigate to the search page and save 2 opportunities by clicking the star icon
7. Navigate to the saved opportunities page and verify 2 saved opportunities
8. On one saved opportunity listing, click on the title to navigate to the opportunity detail view.
9. Verify the saved icon correctly reflects the saved status (yellow)
10. Click to un-save the opportunity and navigate back to the saved opportunities page. Refresh and verify the opportunity is no longer there
11. On the other saved opportunity, click on the "Share with organization" to open the modal and select one of the organizations to save the opportunity to.
12. Now click on that opportunity title to navigate to the opportunity detail page.
13. Verify the saved icon correctly reflects the saved status (yellow)
14. Click the star icon to un-save it and navigate back to the saved opportunities page.
15. Verify you still see the opportunity in the saved opportunity list and verify there is a checked organization in the share modal.
16. Navigate to the search page to find the opportunity in the search results and verify the saved icon does not indicate it is saved
17. Go back to the saved opportunities page and uncheck the organization in the modal and verify the saved opportunity page renders empty

### CODE REVIEW

**Comment by myduong-navapbc** [comment]
i believe we have a typo: `savedOpportunitiiesUtils`, two `ii`s

File: frontend/src/services/fetch/fetchers/savedOpportunityFetcher.ts [hunk 1]
```diff
@@ -80,12 +90,16 @@ export const getSavedOpportunities = async (
   return json.data;
 };
 
-export const getSavedOpportunity = async (
+export const getUserSavedOpportunity = async (
```
Comment (myduong-navapbc) [inline frontend/src/services/fetch/fetchers/savedOpportunityFetcher.ts:93]: I like this, very concise 👍

File: frontend/src/services/fetch/fetchers/savedOpportunityFetcher.ts
Comment (myduong-navapbc) [inline frontend/src/services/fetch/fetchers/savedOpportunityFetcher.ts]: I know you didnt do this part but, we now know what return: 
`as { data: MinimalOpportunity[]`

**Comment by myduong-navapbc** [comment]
Overall this looks really solid 👍 nice work here.

The scope-based approach makes the intent much clearer, and I like how the saved-opportunity check is now explicitly tied to the individual scope instead of relying on the broader saved list. The changes feel well-contained and easy to follow.

The new utility tests are also great - they cover the scope logic well and make the behavior easier to reason about.

Just a couple small things:

Minor typo in `savedOpportunitiiesUtils` filename (extra `i`)
Might be worth adding one test closer to the fetch/handler path to cover the individual-vs-org edge case

Otherwise looks good to me - appreciate the thoughtful implementation here.
